tools/toolchains/__init__.py

```diff
@@ -911,6 +930,7 @@ def build_library(self, objects, dir, name):
 
         return needed_update
 
+    @check_toolchain_path
```

Comment:
We need to test how this decorator works with the existsing `hook_tool` decorator( for example [here](https://github.com/ARMmbed/mbed-os/blob/master/tools/toolchains/gcc.py#L217)). If they don't work together properly, this might break compilation on targets that have a post-link step (for example `LPC4088`).